jib-core/src/main/java/com/google/cloud/tools/jib/ncache/Cache.java

```diff
@@ -16,22 +16,73 @@
 
 package com.google.cloud.tools.jib.ncache;
 
+import com.google.cloud.tools.jib.blob.Blob;
 import com.google.cloud.tools.jib.image.DescriptorDigest;
 import com.google.cloud.tools.jib.image.LayerEntry;
 import com.google.common.collect.ImmutableList;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Optional;
+import javax.annotation.concurrent.Immutable;
 
-/** Cache for storing data to be shared between Jib executions. */
+/**
+ * Cache for storing data to be shared between Jib executions.
+ *
+ * <p>Uses the default cache storage engine ({@link DefaultCacheStorage}) and layer entries as the
+ * selector ({@link LayerEntriesSelector}).
+ *
+ * <p>Implementation is immutable and thread-safe.
+ */
+@Immutable
 public class Cache {
 
+  /**
+   * Initializes the cache using {@code cacheDirectory} for storage.
+   *
+   * @param cacheDirectory the directory for the cache. Creates the directory if it does not exist.
+   * @return a new {@link Cache}
+   * @throws IOException if an I/O exception occurs
+   */
+  public static Cache withDirectory(Path cacheDirectory) throws IOException {
+    Files.createDirectories(cacheDirectory);
+    return new Cache(DefaultCacheStorage.withDirectory(cacheDirectory));
+  }
+
   private final CacheStorage cacheStorage;
 
-  // TODO: Add way to instantiate after #retrieveCacheEntry and #write are implemented.
   private Cache(CacheStorage cacheStorage) {
     this.cacheStorage = cacheStorage;
   }
 
+  /**
+   * Saves a cache entry with only a layer {@link Blob}. Use {@link #write(Blob, DescriptorDigest,
+   * Blob)} to include a selector and metadata.
+   *
+   * @param layerBlob the layer {@link Blob}
+   * @return the {@link CacheEntry} for the written {@link CacheWrite}
+   * @throws IOException if an I/O exception occurs
+   */
+  public CacheEntry write(Blob layerBlob) throws IOException {
+    return cacheStorage.write(CacheWrite.layerOnly(layerBlob));
+  }
+
+  /**
+   * Saves a cache entry with a layer {@link Blob}, an additional selector digest, and a metadata
+   * {@link Blob}. Use {@link #write(Blob)} to save only a layer {@link Blob}.
```

Comment:
I'll actually change this to take `LayerEntry`s instead so that selectors are just an implementation detail of the cache. *Reference* can be an alternative name, but that may be confusing since it is overloaded with both the meaning of *image reference* and the *reference digest*.

Reply:
Sounds good.